### PR TITLE
Updated tests for 'Iterate through an array with a for loop' to ES6

### LINKED ISSUE #16983: "Iterate through an array with a for loop" checks forces you to use "var" cannot use "let" or "const"
#### Challenge Name
https://beta.freecodecamp.org/en/challenges/basic-javascript/iterate-through-an-array-with-a-for-loop

#### Issue Description
The challenge requires you to initialize a variable however it forces you to use var, rather than let or const.

#### Browser Information

* Browser Name, Version: Ungoogled Chromium
* Operating System: macOS
* Mobile, Desktop, or Tablet: Desktop

#### Your Code

```js

N/A

```
#### Screenshot
screen shot 2018-03-29 at 1 23 37 pm

### PR DESCRIPTION
**Pull request by mail-liam**
Updated tests to pass with ES6 syntax. Allows let, const, and no semicolon at end of line.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #16983

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md [hunk 1]
```diff
@@ -22,13 +22,13 @@ Declare and initialize a variable <code>total</code> to <code>0</code>. Use a <c
...
   - text: <code>total</code> should be declared and initialized to 0
-    testString: 'assert(code.match(/var.*?total\s*=\s*0.*?;/), "<code>total</code> should be declared and initialized to 0");'
+    testString: 'assert(code.match(/(var|let|const)\s*?total\s*=\s*0.*?;?/), "<code>total</code> should be declared and initialized to 0");'
   - text: <code>total</code> should equal 20
     testString: 'assert(total === 20, "<code>total</code> should equal 20");'
   - text: You should use a <code>for</code> loop to iterate through <code>myArr</code>
     testString: 'assert(code.match(/for\s*\(/g).length > 1 && code.match(/myArr\s*\[/), "You should use a <code>for</code> loop to iterate through <code>myArr</code>");'
   - text: Do not set <code>total</code> to 20 directly
-    testString: 'assert(!code.match(/total[\s\+\-]*=\s*(\d(?!\s*[;,])|[1-9])/g), "Do not set <code>total</code> to 20 directly");'
+    testString: 'assert(!code.match(/total[\s\+\-]*=\s*[^0].*;?/g), "Do not set <code>total</code> to 20 directly");'
```
Comment (mail-liam) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: This regex inadvertently allows for setting total to 20 through alternate bases. Changing to "/total[\s\+\-]*=\s*([^0]|0[xbo]?\d+)/g" to fix this.

**Comment by mail-liam** [comment]
Finally made the last test pass satisfactorily. Attaching the test cases I used. 

regextestcases

**Comment by scissorsneedfoodtoo** [comment]
@mail-liam, thank you for submitting a PR for this issue. This looks good, though I'd like to test this locally before we merge it.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md [hunk 2]
```diff
@@ -22,13 +22,13 @@ Declare and initialize a variable <code>total</code> to <code>0</code>. Use a <c
...
   - text: <code>total</code> should be declared and initialized to 0
-    testString: 'assert(code.match(/var.*?total\s*=\s*0.*?;/), "<code>total</code> should be declared and initialized to 0");'
+    testString: 'assert(code.match(/(var|let|const)\s*?total\s*=\s*0.*?;?/), "<code>total</code> should be declared and initialized to 0");'
   - text: <code>total</code> should equal 20
     testString: 'assert(total === 20, "<code>total</code> should equal 20");'
   - text: You should use a <code>for</code> loop to iterate through <code>myArr</code>
     testString: 'assert(code.match(/for\s*\(/g).length > 1 && code.match(/myArr\s*\[/), "You should use a <code>for</code> loop to iterate through <code>myArr</code>");'
   - text: Do not set <code>total</code> to 20 directly
-    testString: 'assert(!code.match(/total[\s\+\-]*=\s*(\d(?!\s*[;,])|[1-9])/g), "Do not set <code>total</code> to 20 directly");'
+    testString: 'assert(!code.match(/total[\s+-]*=\s*([^0\s]|0[xbo]?\d+)/g), "Do not set <code>total</code> to 20 directly");'
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: The test is buggy. A solution provided in this file below does not pass the test.

Comment (mail-liam) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: Hi ValeraS. Which file are you referring to?

Comment (ValeraS) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md]: This one which you are editing.

**Comment by mail-liam** [comment]
Updated again. Here are the new tests used:

newfinal

**Comment by scissorsneedfoodtoo** [comment]
Thank you for helping us improve this section of the curriculum, @mail-liam! Looking forward to your next contribution :+1: :+1:

**Pull request merged**